### PR TITLE
Remove code for testing 2.1 providers

### PR DESCRIPTION
**Pull request by bricelam**

### CODE REVIEW

File: test/EFCore.SqlServer.FunctionalTests/EFCore.SqlServer.FunctionalTests.csproj [hunk 1]
```diff
@@ -28,16 +28,12 @@
     </None>
   </ItemGroup>
 
-  <PropertyGroup Condition="'$(FunctionalTests_PackageVersion)' == '2.1.0'">
-   <DefineConstants>$(DefineConstants);Test21</DefineConstants>
-  </PropertyGroup>
-
-  <ItemGroup Condition="'$(FunctionalTests_PackageVersion)' != '2.1.0'">
+  <ItemGroup Condition="'$(FunctionalTests_PackageVersion)' == '0.0.0'">
```
Comment (AndriySvyryd) [inline test/EFCore.SqlServer.FunctionalTests/EFCore.SqlServer.FunctionalTests.csproj]: Why do we still need this?

File: test/EFCore.Sqlite.FunctionalTests/EFCore.Sqlite.FunctionalTests.csproj [hunk 2]
```diff
@@ -17,19 +17,15 @@
...
-  <PropertyGroup Condition="'$(FunctionalTests_PackageVersion)' == '2.1.0'">
-   <DefineConstants>$(DefineConstants);Test21</DefineConstants>
-  </PropertyGroup>
 
   <ItemGroup Condition="'$(FunctionalTests_PackageVersion)' == '0.0.0'">
     <ProjectReference Include="..\..\src\EFCore.Sqlite.NTS\EFCore.Sqlite.NTS.csproj" />
   </ItemGroup>
-  
+
   <ItemGroup Condition="'$(FunctionalTests_PackageVersion)' != '0.0.0'">
```
Comment (smitpatel) [inline test/EFCore.Sqlite.FunctionalTests/EFCore.Sqlite.FunctionalTests.csproj]: Should rename the variable here

File: test/EFCore.SqlServer.FunctionalTests/EFCore.SqlServer.FunctionalTests.csproj (hunk 1 above)
Comment (bricelam) [reply test/EFCore.SqlServer.FunctionalTests/EFCore.SqlServer.FunctionalTests.csproj]: We don’t until we test 3.0 providers on 3.1

Comment (bricelam) [reply test/EFCore.SqlServer.FunctionalTests/EFCore.SqlServer.FunctionalTests.csproj]: Happy to remove. lol, I thought you’d object if I did.

Comment (AndriySvyryd) [reply test/EFCore.SqlServer.FunctionalTests/EFCore.SqlServer.FunctionalTests.csproj]: We'll probably do it in a different way for 3.1.

**Pull request merged**